Norm_end_to_end.Rmd

```diff
@@ -149,41 +183,72 @@ intern_standard$mz <- mapply(intern_standard$formula, intern_standard$POS,
 
 intern_standard$feature_id <- NA_character_
 
-rt_tol <- 30
-for (i in seq_len(nrow(intern_standard))) {
-    fts <- featureDefinitions(data, mz = intern_standard$mz[i], ppm = 50)
-    if (nrow(fts)) {
-        rtdiff <- abs(fts$rtmed - intern_standard$RT[i])
-        keep <- rtdiff < rt_tol
-        if (sum(keep) == 1)
-            intern_standard$feature_id[i] <- rownames(fts)[keep]
-    }
+library(MetaboAnnotation)
```

Comment:
OK, so that's how this can be done with `MetaboAnnotation`. It might seem a bit more difficult, but maybe worth to already introduce. We have plenty of possibilities there to match measured m/z and RT values to theoretical ones and then to handle these matching results (which by default can be n:m).